### PR TITLE
accessibility: issue/#2094 popups

### PR DESCRIPTION
**Pull request by oliverfoster**
#2094

* Siblings of the parent branch of the popup tag become `aria-hidden=true tabindex="-1"`
* Tab index original values are better restored by accounting for no tab index rather than defaulting to `0`

### To test

Using JAWS, enable accessibility mode, open the drawer or a notify. Use the up and down arrow keys to navigation around. You should not be able to read any content behind the popup.

### CODE REVIEW

File: src/core/js/libraries/jquery.a11y.js [hunk 1]
```diff
@@ -1067,6 +1068,10 @@
                 $hideable = $(domSelectors.hideableElements).filter(domFilters.globalTabIndexElementFilter);
             }
 
+            var $branch = this.add(this.parents());
+            var $siblings = $branch.siblings().filter(domFilters.globalTabIndexElementFilter);
```
Comment (tomgreenfield) [inline src/core/js/libraries/jquery.a11y.js]: When we spoke briefly about this, you said you may want to switch to `nextAll()`.

Comment (oliverfoster) [reply src/core/js/libraries/jquery.a11y.js]: this was a different bit of code i was meaning. i misunderstood you.
this bit gets all of the elements which aren't associated with the branch your element exists on. 
such that all other elements other than the branch from root to tip that you need can become excluded to the screen reader.

**Comment by oliverfoster** [comment]
@tomgreenfield could you +1 ?

File: src/core/js/libraries/jquery.a11y.js (hunk 1 above)
Comment (chris-steele) [reply src/core/js/libraries/jquery.a11y.js]: @oliverfoster was struggling to get my head around this, but I think the key is to realise that when the ancestors and their siblings are made aria-hidden then by virtue _their_ descendants are hidden. Hope I've got that right.

Comment (oliverfoster) [reply src/core/js/libraries/jquery.a11y.js]: Exactly. This happens through each parent towards the document. The parent's siblings are made aria hidden. As if picking off all the branches down a twig except the one you need to be available.

**Pull request merged**